backends/dpdk/dpdkArch.cpp

```diff
@@ -2448,4 +2448,242 @@ const IR::Node* ElimHeaderCopy::postorder(IR::Member* m) {
     return m;
 }
 
+cstring EmitDpdkTableConfig::getKeyMatchType(const IR::KeyElement* ke, P4::ReferenceMap* refMap) {
+    auto path = ke->matchType->path;
+    auto mt = refMap->getDeclaration(path, true)->to<IR::Declaration_ID>();
+    BUG_CHECK(mt != nullptr, "%1%: could not find declaration", ke->matchType);
+    return mt->name.name;
+}
+
+int EmitDpdkTableConfig::getTypeWidth(const IR::Type* type,
+                                      P4::TypeMap* typeMap) {
+    return typeMap->widthBits(type, type->getNode(), false);
+}
+
+void EmitDpdkTableConfig::print(cstring str, cstring sep) {
+    dpdkTableConfigFile<<str<<sep;
+}
+
+void EmitDpdkTableConfig::print(big_int str, cstring sep) {
+    dpdkTableConfigFile<<"0x"<<std::hex<<str<<sep;
+}
+
+big_int EmitDpdkTableConfig::convertSimpleKeyExpressionToBigInt(
```

Comment:
Does the added test cover all the if conditions? If not, can we add more entries covering all these cases